Making the declaration for call() an ANSI-style prototype won't work on older
compilers, but then we only need this patch on newer compilers (and it's fixed
in krb5 1.3 anyway, so just go with it).
--- krb5-1.2.8/src/appl/telnet/telnet/commands.c	2000-05-23 20:08:39.000000000 -0400
+++ krb5-1.2.8/src/appl/telnet/telnet/commands.c	2003-07-07 17:12:05.000000000 -0400
@@ -60,7 +60,7 @@
 #include <netdb.h>
 #include <ctype.h>
 #include <pwd.h>
-#include <varargs.h>
+#include <stdarg.h>
 #include <errno.h>
 #ifdef HAVE_VFORK_H
 #include <vfork.h>
@@ -115,7 +115,7 @@
 extern char **genget();
 extern int Ambiguous();
 
-static call();
+static call(int (*)(int, char**), ...);
 
 typedef struct {
 	char	*name;		/* command name */
@@ -2698,17 +2698,13 @@
 
     /*VARARGS1*/
     static
-call(va_alist)
-    va_dcl
+call(int (*routine)(int, char **), ...)
 {
     va_list ap;
-    typedef int (*intrtn_t)();
-    intrtn_t routine;
     char *args[100];
     int argno = 0;
 
-    va_start(ap);
-    routine = (va_arg(ap, intrtn_t));
+    va_start(ap, routine);
     while ((args[argno++] = va_arg(ap, char *)) != 0) {
 	;
     }
